herbert_core/classes/instrument_classes/instrument/@IX_null_sample/IX_null_sample.m

```diff
@@ -9,12 +9,258 @@
+            %
+            %
+            % See also structIndep, structArrPublic, structArrIndep
+            
+            names = obj.propNamesPublic_';
+            if ~isempty(obj)
+                tmp = obj(1);
+                S = obj.scalarEmptyStructPublic_;
+                for i=1:numel(names)
+                    S.(names{i}) = tmp.(names{i});
+                end
+            else
+                args = [names; repmat({cell(size(obj))},size(names))];
+                S = struct(args{:});
+            end
+        end
+        
+        function S = structArrPublic(obj)
+            % Return the public properties of an object array as a structure array
+            %
+            %   >> s = structArrPublic(obj)
+            %
+            % Use <a href="matlab:help('structPublic');">structPublic</a> for behaviour that more closely matches the Matlab
+            % intrinsic function struct.
+            %
+            % Has the same behaviour as the Matlab instrinsic struct in that:
+            % - Any structure array is returned unchanged
+            % - If an object is empty, an empty structure is returned with fieldnames
+            %   but the same size as the object
+            %
+            % However, differs in the behaviour if an object array:
+            % - If the object is non-empty array, returns a structure array of the same
+            %   size. This is different to the instrinsic Matlab, which returns a scalar
+            %   structure from the first element in the array of objects
+            %
+            %
+            % See also structPublic, structIndep, structArrIndep
+            
+            if numel(obj)>1
+                S = arrayfun(@fill_it, obj);
+            else
+                S = structPublic(obj);
+            end
+            
+            function S = fill_it (obj)
+                names = obj.propNamesPublic_';
+                S = obj.scalarEmptyStructPublic_;
+                for i=1:numel(names)
+                    S.(names{i}) = obj.(names{i});
+                end
+            end
+
+        end
+    end
+    
+    %======================================================================
+    % Custom loadobj and saveobj
+    % - to enable custom saving to .mat files and bytestreams
+    % - to enable older class definition compatibility
+
+    methods
+        %------------------------------------------------------------------
+        function S = saveobj(obj)
+            % Method used my Matlab save function to support custom
+            % conversion to structure prior to saving.
+            %
+            %   >> S = saveobj(obj)
+            %
+            % Input:
+            % ------
+            %   obj     Scalar instance of the object class
+            %
+            % Output:
+            % -------
+            %   S       Structure created from obj that is to be saved
+            
+            % The following is boilerplate code
```

Comment:
To avoid issues with changes in a future, the objects should be saved with version number.
not sure what is better -- specify constant property `version =1 `  (will `structIndep` pick it up?)
or add here
```
S.version =1
```
loadobj should strip the version accordingly (and process data according to version)